File: scala/raindrops/src/main/scala/Raindrops.scala
object Raindrops {
  private val drops = Map(3 -> "Pling", 5 -> "Plang", 7 -> "Plong")

  def convert(n: Int): String = {
    val sounds = for ((factor, sound) <- drops if n % factor == 0)
                    yield sound

    if (sounds.isEmpty) n.toString else sounds.mkString
  }
}

File: kotlin/raindrops/src/main/kotlin/Raindrops.kt
object Raindrops {
    private val factors = mapOf(3 to "Pling", 5 to "Plang", 7 to "Plong")

    fun convert(n: Int) =
        factors.entries
            .filter { n % it.key == 0 }
            .joinToString("") { it.value }
            .ifEmpty { n.toString() }
}